File: ch02/7用两个栈实现队列/stackToQueue.py
    # -*- coding:utf-8 -*-
# python中是用list来表示栈这种数据结构的
class Solution:
    def __init__(self):
        self.stack1 = []
        self.stack2 = []

    def push(self, node):
        # write code here
        while len(self.stack2) != 0:
            tmp = self.stack2.pop()
            self.stack1.append(tmp)
        self.stack1.append(node)

    def pop(self):
        # return xx
        while len(self.stack1) != 0:
            tmp = self.stack1.pop()
            self.stack2.append(tmp)
        return self.stack2.pop()

File: ch02/7用两个栈实现队列/stackToQueue.cpp
class Solution
{
public:
    void push(int node) {
        if(stack2.empty()) {
            stack1.push(node);
        }
        else {
            while(!stack2.empty()) {
                stack1.push(stack2.top());
                stack2.pop();
            }
            stack2.push(node);
        }
        
    }

    int pop() {
        while(!stack1.empty()) {
            stack2.push(stack1.top());
            stack1.pop();
        }
        int re = stack2.top();
        stack2.pop();
        return re;
        
    }

private:
    stack<int> stack1;
    stack<int> stack2;
};